Marlin/src/gcode/feature/pause/M125.cpp

```diff
@@ -96,10 +96,10 @@ void GcodeSuite::M125() {
   const bool show_lcd = TERN0(HAS_MARLINUI_MENU, parser.boolval('P'));
 
   if (pause_print(retract, park_point, show_lcd, 0)) {
-    if (ENABLED(EXTENSIBLE_UI) || BOTH(EMERGENCY_PARSER, HOST_PROMPT_SUPPORT) || !sd_printing || show_lcd) {
+    if (ANY(HAS_DISPLAY, HAS_LCDPRINT) || BOTH(EMERGENCY_PARSER, HOST_PROMPT_SUPPORT) || !sd_printing || show_lcd) {
       wait_for_confirmation(false, 0);
-      resume_print(0, 0, -retract, 0);
     }
+    resume_print(0, 0, -retract, 0);
```

Comment:
This change seems problematic. It would seem to resume the print immediately upon pausing it for the other cases — those where this is not the whole pause/resume handler. The idea of the `wait...` and `resume...` being handled here is that this block is the entire synchronous handler of pausing under the given conditions. For the others, this command will exit, and the wait and the resume should both be happening elsewhere. We can trace this all out if needed, and look again at the history of these changes to see where the wait and the resume are expected to be handled. But in any case, we should definitely not be resuming unless we are also waiting.

Reply:
You are right, but without this call printer does an half resume (return to print position for sure and not, so sure I recall it correctly, it does a small "continue").
With this call and with my configuration printer ask to purge or continue due to `load_filament` call.

Reply:
for a safer case we may put `resume_print` back into if scope. I don't like the idea that print hangs after a pasuse but when there is no other way to resume it may be one of the possible choices

Reply:
I'd recommend making a full flowchart of the various pause functions before making further adjustments. Maybe the whole system can be cleaned up.